presentation/matrix_view.py

```diff
@@ -4,72 +4,124 @@
-    data = []
-    for row in mesh.rows:
-        row_data = {"agent_alias": row.alias}
+def make_data(mesh: MeshResults, matrix: Matrix, latency_deteriorated_ms: int, latency_failed_ms: int) -> List[Dict]:
+    x = matrix.agents
+    y = [i for i in reversed(matrix.agents)]
+    z = []
+
+    for row in reversed(mesh.rows):
+        z_col = []
         for col in row.columns:
-            latency_ms = matrix.cells[row.alias][col.alias].latency_microsec.value / 1000  # scale to milliseconds
-            row_data[col.alias] = f"{latency_ms} ms"
-        data.append(row_data)
+            color = get_color(
+                matrix.cells[row.alias][col.alias].latency_microsec.value, latency_deteriorated_ms, latency_failed_ms
+            )
+            z_col.append(color)
+        z.append(z_col)
+    for i in range(len(mesh.rows)):
+        z[-(i + 1)].insert(i, None)
+    z = np.array(z)
+
+    colorscale = get_colorscale(z)
+
+    data = [
+        dict(
+            x=x,
+            y=y,
+            z=z,
+            text=make_hover_text(mesh, matrix),
+            type="heatmap",
+            hoverinfo="text",
+            opacity=1,
+            name="",
+            showscale=False,
+            colorscale=colorscale,
+        )
+    ]
+
     return data
 
 
-def make_colors(mesh: MeshResults, latency_deteriorated_ms: int, latency_failed_ms: int):
-    styles = []
-    for index, row in enumerate(mesh.rows):
-        for column in row.columns:
-            if column.latency_microsec.value > latency_failed_ms * 1000:
-                styles.append(
-                    {
-                        "if": {
-                            "column_id": column.alias,
-                            "row_index": index,
-                        },
-                        "backgroundColor": "red",
-                    }
-                )
-            elif column.latency_microsec.value > latency_deteriorated_ms * 1000:
-                styles.append(
-                    {
-                        "if": {
-                            "column_id": column.alias,
-                            "row_index": index,
-                        },
-                        "backgroundColor": "orange",
-                    }
-                )
-            else:
-                styles.append(
-                    {
-                        "if": {
-                            "column_id": column.alias,
-                            "row_index": index,
-                        },
-                        "backgroundColor": "green",
-                    }
+def make_annotations(mesh: MeshResults, matrix: Matrix) -> List[Dict]:
+    annotations = []
+    for row in reversed(mesh.rows):
+        for col in row.columns:
+            annotations.append(
+                dict(
+                    showarrow=False,
+                    text=f"<b>{(matrix.cells[row.alias][col.alias].latency_microsec.value * 1e-3):.2f} ms<b>",
```

Comment:
```suggestion
                    text=f"<b>{(matrix.cells[row.alias][col.alias].latency_microsec.value * 1e-3):.2f} ms</b>",
```
I guess this is an HTML tag

Reply:
Yes you are right. I fixed this typo

Reply:
Done